R/github.R

```diff
@@ -537,3 +540,20 @@ get_gh_contrib_issue <- function (org, repo) {
 
     unlist (pings)
 }
+
+#' Check the GitHub rate limit and warn if exceeded.
+#'
+#' @noRd
+check_rate_limit <- function () {
+    gh_state <- gh::gh_rate_limit()
+    if (gh_state$remaining == 0) {
```

Comment:
Just a suggestion, but how about an additional check like this:
``` r
limit_warn <- 0.1
warn_txt <- NULL
if (gh_state$remaining / gh_state$limit < limit_warn) {
  warn_txt <- "You have used > 90% of your GitHub calls..."
} else if (gh_state$remaining == 0) {
  warn_txt <- "The GitHub rate limit has been reached ..."
}
if (!is.null (warn_txt)) {
  cli::cli_alert_warning (warn_txt, wrap = FALSE)
}
```
GitHub itself now warns on several things at a limit of 75%, but I think for rate calls which reset every hour that 90% should be fine. Oh, and we should definitely also mention that it is an *hourly* limit.